tarofreighter/chain_porter.go

```diff
@@ -1160,6 +1168,21 @@ func (p *ChainPorter) publishSubscriberEvent(event Event) {
 // chanutils.EventPublisher interface.
 var _ chanutils.EventPublisher[Event, bool] = (*ChainPorter)(nil)
 
+// ExecuteSendStateEvent is an event which is sent to the ChainPorter's event
+// subscribers before a state is executed.
+type ExecuteSendStateEvent struct {
+	// timestamp is the time the event was created.
+	timestamp time.Time
```

Comment:
Since this is non-exported, should we add a `NewExecuteSendStateEvent` function that sets the timestamp to now and one `NewExecuteSendStateEventAt` that allows you to set a custom timestamp?

Reply:
I'll add `NewExecuteSendStateEvent`.

I don't have a use case for `NewExecuteSendStateEventAt` at the moment. I'll leave it out for that reason, unless you think I should add it anyway for convention or something like that, please let me know if that's the case.